lib/components/map/base-map.js

```diff
@@ -52,46 +88,78 @@ class BaseMap extends Component {
     // }
   }
 
-  componentWillReceiveProps (nextProps) {
+  _updateBounds (oldProps, newProps) {
     // TODO: maybe setting bounds ought to be handled in map props...
 
+    const { map } = this.refs
+    if (!map) return
+
+    const padding = [30, 30]
+
     // Fit map to to entire itinerary if active itinerary bounds changed
-    const oldItinBounds = this.props.itinerary && getItineraryBounds(this.props.itinerary)
-    const newItinBounds = nextProps.itinerary && getItineraryBounds(nextProps.itinerary)
+    const oldItinBounds = oldProps && oldProps.itinerary && getItineraryBounds(oldProps.itinerary)
+    const newItinBounds = newProps.itinerary && getItineraryBounds(newProps.itinerary)
     if (
       (!oldItinBounds && newItinBounds) ||
       (oldItinBounds && newItinBounds && !oldItinBounds.equals(newItinBounds))
     ) {
-      this.refs.map &&
-        this.refs.map.leafletElement.fitBounds(newItinBounds, {
-          padding: [3, 3]
-        })
-    }
+      map.leafletElement.fitBounds(newItinBounds, { padding })
+
+    // If no itinerary present but from/to locations are, fit to those
+    } else if (newProps.query.from && newProps.query.to) {
+      map.leafletElement.fitBounds([
+        [newProps.query.from.lat, newProps.query.from.lon],
+        [newProps.query.to.lat, newProps.query.to.lon]
+      ], { padding })
+
+    // If only from or to is set, pan to that
+    } else if (newProps.query.from) {
+      map.leafletElement.panTo([newProps.query.from.lat, newProps.query.from.lon])
+    } else if (newProps.query.to) {
+      map.leafletElement.panTo([newProps.query.to.lat, newProps.query.to.lon])
```

Comment:
We should be avoiding direct calls to the leaflet map whenever possible and instead send props to the Leaflet map

Reply:
To be addressed in #44.